File: leetcode/power-of-two.rs
/*

Given an integer, write a function to determine if it is a power of two.

*/

fn main() {
    for i in 1..(1 << 20) + 1 {
        if is_power_of_two(i) {
            println!("{}", i);
        }
    }
}

fn is_power_of_two(n: usize) -> bool {
    n > 0 && (n & (n - 1)) == 0
}

File: codewars/power-of-two.py
#!/usr/bin/env python3

"""

Complete the function power_of_two/powerOfTwo (or equivalent, depending on your language) that determines if a given non-negative integer is a power of two. From the corresponding Wikipedia entry:

a power of two is a number of the form 2n where n is an integer, i.e. the result of exponentiation with number two as the base and integer n as the exponent.

You may assume the input is always valid.

Examples
isPowerOfTwo(1024) // -> true
isPowerOfTwo(4096) // -> true
isPowerOfTwo(333)  // -> false
Beware of certain edge cases - for example, 1 is a power of 2 since 2^0 = 1 and 0 is not a power of 2.

"""

def ispow2(n):
    return n > 0 and n&(n-1) == 0

def main():
    for i in range(1<<21 + 1):
        if ispow2(i):
            print(i)

main()
